### PR TITLE
Change phase in Cosine1D and Sine1D to phase shift [ci skip]

### LINKED ISSUE #16470: Describe "phase" parameter in astropy.modeling.functional_models.Sine1D and astropy.modeling.functional_models.Cosine1D
### What is the problem this feature will solve?

The documentation for astropy.modeling.functional_models.Sine1D should describe the units of the  "phase" parameter. Since it does not, its not clear to the user whether that parameter has the same units of the x-axis (which I would call an x-offset, not a phase), or is it phase in the sense of an angular offset (units of degrees or radians).
 

### Describe the desired outcome

Add description of the "phase" parameter to the documentation of Cosine1D and Sine1D, including units

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Parkerwise**
### Description

This pull request is to address the documentation for astropy.modeling.functional_models.Cosine1D. The word "phase" was changed to "phase shift" to indicate that the variable is referring to an x-offset. This change was also made for Sine1D.

Fixes #16470 

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by neutrinoceros — COMMENTED** [review]
Hello again, thank you for your PR ! I'll let others review the actual content but I'd like to point out that you can't really use `[ci skip]` if you don't intend to push more commits, because CI is actually required to run on any PR before we can merge it.
Here's how to amend it:
```
GIT_EDITOR=<name_of_your_favorite_editor_here> git commit --amend
# ...
# update commit message, removing [ci skip], save and quit (get back to terminal)
# ...
git push --force
```

**Comment by pllim** [comment]
Re: ci skip

Since this is trivial doc change that does not touch doctest, `[ci skip]` is allowed. Just need admin override to merge after approval, which I can do. FYI. Thanks!

**Review by WilliamJamieson — APPROVED** [review]

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**
